sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -245,9 +247,9 @@ internal static RecognizeLinkedEntitiesResultCollection ConvertToRecognizeLinked
 
         #region Healthcare
 
-        internal static IReadOnlyCollection<HealthcareEntity> ConvertToHealthcareEntityCollection(IEnumerable<HealthcareEntityInternal> healthcareEntities, IEnumerable<HealthcareRelationInternal> healthcareRelations)
+        internal static List<HealthcareEntity> ConvertToHealthcareEntityCollection(IEnumerable<HealthcareEntityInternal> healthcareEntities, IEnumerable<HealthcareRelationInternal> healthcareRelations)
         {
-            return healthcareEntities.Select((entity) => new HealthcareEntity(entity, healthcareEntities, healthcareRelations)).ToList();
+            return healthcareEntities.Select((entity) => new HealthcareEntity(entity, ResolveRelatedEntities(entity, healthcareEntities, healthcareRelations))).ToList();
```

Comment:
This is getting close. Definitely better to have the code here in the Transformation class. the O(n2) complexity is still a problem.
If you follow what JS did, it will look something like:
```
var entities = healthcareEntities.Select((entity) => new HealthcareEntity(entity)).ToList(); //no relationship assigned
// then u assign the relationships. This code is in JS but u can map it to .NET
for (const relation of healthcareRelations) {
    const relationType = relation.relationType;
    const sourceIndex = parseHealthcareEntityIndex(relation.source); // u can have a function which is in charge of doing the regex
    const targetIndex = parseHealthcareEntityIndex(relation.target);
    const sourceEntity = entities[sourceIndex];
    const targetEntity = entities[targetIndex];
    sourceEntity.relatedEntities.Add(targetEntity, relationType);
      if (relation.bidirectional) {
        targetEntity.relatedEntities.Add(sourceEntity, relationType);
      }
}
return entities;
```

In order for this to work you will have to instantiate the `RelatedEntities` by default in `HealthcareEntity` so something like:
```
public IReadOnlyDictionary<HealthcareEntity, HealthcareEntityRelationType> RelatedEntities { get; } = new Dictionary<HealthcareEntity, HealthcareEntityRelationType>();
```